src/test/scala/pick16SCandidates.scala

```diff
@@ -104,19 +104,49 @@ case object pick16SCandidates extends FilterData(
     ( (seq.count(_ == 'N') / seq.length) <= 0.01 )
   }
 
+  // TODO: move it to db.rnacentral
+  implicit class IteratorOps[V](val iterator: Iterator[V]) extends AnyVal {
+
+    /* Similar to the Stream's .groupBy, but assuming that groups are contiguous. Another difference is that it returns the key corresponding to each group. */
+    // NOTE: The original iterator should be discarded after calling this method
+    def groupBy[K](getKey: V => K): Iterator[(K, Seq[V])] = new Iterator[(K, Seq[V])] {
+      /* The definition is very straightforward: we keep the `rest` of values and on each `.next()` call bite off the longest prefix with the same key */
+
+      /* Buffered iterator allows to loo ahead without removing the next element */
+      private val rest: BufferedIterator[V] = iterator.buffered
```

Comment:
[Not much](https://github.com/scala/scala/blob/v2.11.8/src/library/scala/collection/Iterator.scala#L991-L1011). Do you really think it should be lazy?